### PR TITLE
Show funder changes in log, and refactor `ResourceCompareService` so it will pick up new properties by default

### LINKED ISSUE #867: Show funder changes in change log
To reproduce, change the name of a funder. I would expect to see this in the change log, but I don't.

## Acceptance Criteria

- [ ] Change the change log to include funding
- [ ] Change the system test that modifies the funder to make sure the change log gets updated when the funder changes

### LINKED ISSUE #874: Verify the change log for every field in the Edit form
Write a new system test that verifies that when a user modifies a field in the edit form the change log is update appropriately.

If there are fields that do not update the change log, write a pending test and ticket the update.

Acceptance criteria:
- [ ] There is a system test for the change log after user edits for all field on the edit form for a regular user
- [ ] There is a system test for the change log after user edits for all field on the edit form for a curator user that include the curator controlled fields
- [ ] There is a pending test that documents any fields that are not working correctly and associated tickets if needed

### PR DESCRIPTION
**Pull request by mccalluc**
- fix #867
- fix #874 

One caveat: There are form fields that don't populate the resource, ie, notes to curator. I don't think they are in scope, but I could be wrong.

### CODE REVIEW

File: app/services/resource_compare_service.rb [hunk 1]
```diff
@@ -32,52 +32,49 @@ def identical?
...
-        compare_simple_values(field)
-      end
-      compare_rights
-      [:keywords, :collection_tags].each do |field|
-        compare_arrays(field)
+      @before.as_json.keys.map(&:to_sym).each do |method_sym|
+        before_value = @before.send(method_sym)
+        after_value = @after.send(method_sym)
+        if before_value.is_a?(Array)
+          next if before_value.empty? && after_value.empty?
```
Comment (carolyncole) [inline app/services/resource_compare_service.rb:39]: I wonder if you could use equality to exit early for all cases and simplify the code a bit.
Aka on line 37 you could put `next if before_value.to_json == after_value.to_json`
I did a bit of fiddling in the console and nil.to_json is equivalent to nil.to_json and also the full object equivalence will work with a to_json.
```
[14] pry(main)> before = PDCMetadata::Resource.new(doi:"abc123")
=> #<PDCMetadata::Resource:0x00000001155940a0
 @ark=nil,
 @collection_tags=[],
 @contributors=[],
 @creators=[],
 @description=nil,
 @doi="abc123",
 @funders=[],
 @keywords=[],
 @publication_year=2023,
 @publisher="Princeton University",
 @related_objects=[],
 @resource_type="Dataset",
 @resource_type_general="Dataset",
 @rights=nil,
 @titles=[],
 @version_number="1">
[15] pry(main)> after = PDCMetadata::Resource.new(doi:"abc123")
=> #<PDCMetadata::Resource:0x0000000114e15630
 @ark=nil,
 @collection_tags=[],
 @contributors=[],
 @creators=[],
 @description=nil,
 @doi="abc123",
 @funders=[],
 @keywords=[],
 @publication_year=2023,
 @publisher="Princeton University",
 @related_objects=[],
 @resource_type="Dataset",
 @resource_type_general="Dataset",
 @rights=nil,
 @titles=[],
 @version_number="1">
[16] pry(main)> before == after
=> false
[17] pry(main)> before.to_json == after.to_json
=> true
[18] pry(main)> before = nil
=> nil
[19] pry(main)> after = nil
=> nil
[20] pry(main)> before.to_json == after.to_json
```

File: spec/services/resource_compare_service_spec.rb [hunk 2]
```diff
@@ -54,4 +55,112 @@
...
+        expect(differences).to eq nil
+      end
+
+      it "does flag integer->string+1 as change" do
+        compare = described_class.new(MockResource.new({ fake_year: 2000 }), MockResource.new({ fake_year: "2001" }))
+        differences = compare.differences[:fake_year]
+        expect(differences).to eq [{ action: :changed, from: "2000", to: "2001" }]
+      end
+
+      it "does flag nil->string as change" do
```
Comment (carolyncole) [inline spec/services/resource_compare_service_spec.rb:132]: I believe this is string->nil
```suggestion
      it "does flag string->nil as change" do
```

File: spec/services/resource_compare_service_spec.rb [hunk 3]
```diff
@@ -54,4 +55,112 @@
...
+        compare = described_class.new(MockResource.new({ fake_year: "2000" }), MockResource.new({ fake_year: nil }))
+        differences = compare.differences[:fake_year]
+        expect(differences).to eq [{ action: :changed, from: "2000", to: "" }]
+      end
+
+      it "does flag string->nil as change" do
```
Comment (carolyncole) [inline spec/services/resource_compare_service_spec.rb:138]: I believe this is nil->string
```suggestion
      it "does flag nil->string as change" do
```

**Review by carolyncole — CHANGES_REQUESTED** [review]
Overall this code is tested well.  I put in 3 concrete suggestions that I would like to see changed.
The other suggestion is not a requirement, just a thought to possibly simplify the code a bit.

File: app/services/resource_compare_service.rb (hunk 1 above)
Comment (mccalluc) [reply app/services/resource_compare_service.rb:39]: Sure! Let me merge the other suggestions, and then I'll add that locally.

**Review by carolyncole — APPROVED** [review]

**Pull request merged**